src/PeopleSDKAdapter.js

```diff
@@ -0,0 +1,107 @@
+
+/**
+ * The `PeopleSDKAdapter` is an implementation of the `PeopleAdapter` interface.
+ * This adapter utilizes the Webex JS SDK to fetch data about a person.
+ *
+ * @class PeopleSDKAdapter
+ * @extends {PeopleAdapter}
+ */
+export default class PeopleSDKAdapter extends PeopleAdapter {
+  constructor(datasource) {
+    super(datasource);
+
+    this.getPersonObservables = {};
+  }
+
+  /**
+   * Returns a proper Enum key for a given PersonStatus value
+   *
+   * @private
+   * @param {string} status
+   * @returns {string} PersonStatus
+   * @memberof PeopleSDKAdapter
+   */
+  getStatus(status) {
+    return Object.keys(PersonStatus).find((key) => PersonStatus[key] === status);
+  }
+
+  /**
+   * Fetches the person data from the sdk and returns in the shape required by adapter.
+   *
+   * @private
+   * @param {string} ID  ID of the person for which to fetch data
+   * @returns {Person}
+   * @memberof PeopleSDKAdapter
+   */
+  async fetchPerson(ID) {
+    const {id, emails, displayName, firstName, lastName, avatar, orgId} = await this.datasource.people.get(ID);
+
+    return {
+      ID: id,
+      emails,
+      displayName,
+      firstName,
+      lastName,
+      avatar,
+      orgID: orgId,
+    };
+  }
+
+  /**
+   * Returns an observable that emits person data of the given ID.
+   *
+   * @public
+   * @param {string} ID ID of person to get.
+   * @returns {Observable.<Person>}
+   * @memberof PeopleSDKAdapter
+   */
+  getPerson(ID) {
+    if (!(ID in this.getPersonObservables)) {
+      const personUUID = deconstructHydraId(ID).id;
+      const person$ = from(this.fetchPerson(ID));
+
+      // Subscribe to `Apheleia` internal service to listen for presence changes
+      // And update the person object with status response from the subscription
+      const personWithStatus$ = from(this.datasource.internal.presence.subscribe(personUUID)).pipe(
+        map((data) => data.responses[0].status.status),
+        flatMap((status) => person$.pipe(map((person) => ({...person, status: this.getStatus(status)}))))
+      );
+
+      // Listen for presence changes for the given person ID after subscription to service
+      const presenceUpdate$ = fromEvent(this.datasource.internal.mercury, USER_PRESENCE_UPDATE_EVENT).pipe(
+        filter((event) => event.data.subject === personUUID),
+        map((event) => this.getStatus(event.data.status))
+      );
+
+      // Update the person status after each change emitted from the event
+      const personUpdate$ = person$.pipe(
+        flatMap((person) => presenceUpdate$.pipe(map((status) => ({...person, status}))))
+      );
+
+      // Fetch the original person data on the first run
+      // and updated object after each status change
+      const getPerson$ = concat(personWithStatus$, personUpdate$).pipe(
+        finalize(() => {
+          // Unsubscribe from `Apheleia` internal service
+          // and remove the targeted connectedObservable
```

Comment:
Can you remove this line? It's misleading (and untrue) as there are no connected observables at this point.
Comment still flows without this

Reply:
it is true but yeah sure.

Reply:
There are no connected observables here, it's just a regular observable. A `ConnectedObservable` only gets created by `publishReplay`, all other operators return a regular `Observable`.

Reply:
okay sure lalli. Is there anything else?